baekjoon/divide_and_conquer/17829.js

```diff
@@ -0,0 +1,42 @@
+const solution = (length, matrix) => {
+  const pooling = (length, matrix) => {
+    if (length === 1) return matrix[0][0];
+
+    const half = length / 2;
+    const pooled = Array(half)
+      .fill(null)
+      .map(() => Array(half).fill(0));
+
+    for (let i = 0; i < half; i++) {
+      for (let j = 0; j < half; j++) {
+        const pool = [];
+        pool.push(
+          matrix[i * 2][j * 2],
+          matrix[i * 2 + 1][j * 2],
+          matrix[i * 2][j * 2 + 1],
+          matrix[i * 2 + 1][j * 2 + 1]
+        );
```

Comment:
```suggestion
    for (let i = 0; i < length; i+=2) {
      for (let j = 0; j < length; j+=2) {
        const pool = [];
        pool.push(
          matrix[i][j],
          matrix[i + 1][j],
          matrix[i][j + 1],
          matrix[i + 1][j + 1]
        );
```
위 로직을 다른 함수로 분리하는 것도 좋을 것 같습니다.

Reply:
```suggestion
    const getSecond =(i, j, matrix) => {
         const pool = [];
        pool.push(
          matrix[i * 2][j * 2],
          matrix[i * 2 + 1][j * 2],
          matrix[i * 2][j * 2 + 1],
          matrix[i * 2 + 1][j * 2 + 1]
        );
        
        pool.sort((a,b)=> a-b);
        return pool[2];
    }
    for (let i = 0; i < half; i++) {
      for (let j = 0; j < half; j++) {
          pooled[i][j] = getSecond(i, j, pool);
          }
      }
```
네 이런 구조가 되면 더 깔끔해겠네요 ㅎㅎ.
인덱스는 반복문의 인덱스를 pooled 의 인덱스에 맞춰서 표현하는게, 개인적으로 더 이해하기 쉽다고 느껴집니다.